contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,480 @@
+```
+
+## QoS level cgroups
+
+The `kubelet` defines a `--cgroup-root` flag that is used to specify the `ROOT`
+node in the cgroup hierarchy below which the `kubelet` should manange individual
+cgroup sandboxes.  It is strongly recommended that users keep the default
+value for `--cgroup-root` as `/` in order to avoid deep cgroup hierarchies.  If
+`--enforce-node-alloctable` is enabled per [node allocatable](node-allocatable.md),
+the `ROOT` cgroup sandbox is `ROOT/kubepods`.  For simplicity, we will continue
+to refer to that as `ROOT` in this document.
+
+The `ROOT` cgroup sandbox is used to parent all pod sandboxes that are in
+the Guaranteed QoS class.  By definition, pods in this class have cpu and 
+memory limits specified that are equivalent to their requests so the pod 
+level cgroup sandbox confines resource consumption without the need of an
+additional cgroup sandbox for the tier.
+
+When the `kubelet` launches, it will ensure a `Burstable` cgroup sandbox
+and a `BestEffort` cgroup sandbox exist as children of `ROOT`.  These cgroup
+sandboxes will parent pod level cgroups in those associated QoS classes.
+
+The `kubelet` highly prioritizes resource utilization, and thus
+allows BestEffort and Burstable pods to potentially consume as many
+resources that are presently available on the node.
+
+For compressible resources like CPU, the `kubelet` attempts to mitigate
+the issue via its use of CPU CFS shares.  CPU time is proportioned
+dynamically when there is contention using CFS shares that attempts to
+ensure minimum requests are satisfied.
+
+For incompressible resources, this prioritization scheme can inhibit the
+ability of a pod to have its requests satisfied.  For example, a Guaranteed
+pods memory request may not be satisfied if there are active BestEffort
+pods consuming all available memory.
+
+As a node operator, I may want to satisfy the following use cases:
+
+1. I want to prioritize access to compressible resources for my system
+and/or kubernetes daemons over end-user pods.
+1. I want to prioritize access to compressible resources for my Guaranteed
+workloads over my Burstable workloads.
+1. I want to prioritize access to compressible resources for my Burstable
+workloads over my BestEffort workloads.
+
+Almost all operators are encouraged to support the first use case by enforcing
+[node allocatable](node-allocatable.md) via `--system-reserved` and `--kube-reserved`
+flags.  It is understood that not all operators may feel the need to extend
+that level of reservation to Guaranteed and Burstable workloads if they choose
+to prioritize utilization.  That said, many users in the community deploy
+cluster services as Guaranteed workloads via a `DaemonSet` and would like a similar
+resource reservation model as is provided via [node allocatable](node-allocatable)
+for system and kubernetes daemons.  In addition, depending how operators choose
+to leverage BestEffort workloads, they may choose to make a similar decision
```

Comment:
This rationale calls for having separate configuration for `G` and `Bu` QoS levels